docs/api-reference/extensions/clip-extension.md

```diff
@@ -0,0 +1,59 @@
+
+# ClipExtension
+
+The `ClipExtension` adds support for clipping rendered layers by rectangular bounds.
+
+```js
+import {BitmapLayer} from '@deck.gl/layers';
+import {ClipExtension} from '@deck.gl/extensions';
+
+const layer = new BitmapLayer({
+  ...
+  extensions: [new ClipExtension()],
+  clipBounds: [-180, -85.051129, 180, 85.051129],
+});
+```
+
+## Installation
+
+To install the dependencies from NPM:
+
+```bash
+npm install deck.gl
+# or
+npm install @deck.gl/core @deck.gl/layers @deck.gl/extensions
+```
+
+```js
+import {ClipExtension} from '@deck.gl/extensions';
+new ClipExtension({});
+```
+
+To use pre-bundled scripts:
+
+```html
+<script src="https://unpkg.com/deck.gl@^7.0.0/dist.min.js"></script>
+<!-- or -->
+<script src="https://unpkg.com/@deck.gl/core@^7.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/@deck.gl/layers@^7.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/@deck.gl/extensions@^7.0.0/dist.min.js"></script>
+```
+
+```js
+new deck.ClipExtension({});
+```
+
+## Constructor
+
+```js
+new ClipExtension();
+```
+
+## Layer Properties
+
+When added to a layer via the `extensions` prop, the `ClipExtension` requires the `clipBounds` prop of the layer to contain the rectangular bounds to be used for clipping the rendered region.
```

Comment:
Each prop should have its own header, follow the convention in https://github.com/visgl/deck.gl/blob/master/docs/api-reference/extensions/data-filter-extension.md#layer-properties